File: demo/python/WorkerB.py
import time
from datetime import datetime
from Worker import Worker, WorkerOutput

class WorkerB(Worker):
  def __init__(self):
    super().__init__()
    self.name = "Worker B"
    self.channel = "worker_b"

  def execute_job (self, input : map, parents : map) -> WorkerOutput:
    if "throw" in input:
      print("~~ Worker B Throw!")
      raise Exception(input["throw"])
    else:
      time.sleep(2)
      return WorkerOutput(output={"message": "Worker A did it!", "at": str(datetime.now()) })

  def cleanup (self):
    print("Nothing to cleanup for worker " + self.name + " (" + self.id + ")")

  def is_healthy(self):
    return True

File: demo/js/WorkerB.ts
import Worker from '../../src/Worker'
import TaskResponse from '../../src/TaskResponse'

// This worker is here as an example, and for testing only - do not deploy

export default class WorkerB extends Worker {
  name = "Worker B"
  channel = "worker_b"

  public async executeTask (data: any): Promise<TaskResponse> {
    if (data && data.throw) {
      throw new Error(data.throw)
    } else {
      await new Promise((resolve, reject) => {setTimeout(resolve, 2000)})
      return {output: {"message": "Worker B did it!", "at": new Date() + ''}}
    }
  }
}
